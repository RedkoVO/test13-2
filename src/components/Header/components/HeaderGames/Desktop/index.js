import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'

import UserImg from '../../../../../assets/images/user.jpg'

import styles from './styles'

const HeaderGames = ({ classes, children }) => (
  <div className={classes.root}>
    <div className={classes.gamesUserBlock}>
      <p className={classes.gamesUserHello}> Hello, Ben A <br />
        <Link to={'/'} className={classes.gamesUserVerification}>KYC VERIFICATION</Link>
      </p>
      <div className={classes.gamesUserPhoto}>
        <img src={UserImg} alt="" />
      </div>
    </div>
    
    {children}
  </div>
)

HeaderGames.propTypes = {
  classes: PropTypes.object,
  children: PropTypes.node
}

export default withStyles(styles)(HeaderGames)