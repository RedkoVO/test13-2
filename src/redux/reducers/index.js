import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { APP_RESET } from '../actions/types'
import auth from './auth'
import balance from './balance'
import config from './config'
import history from './history'

const appReducer = combineReducers({
  form,
  auth,
  balance,
  config,
  history
})

export default (state, action) => {
  if (action.type === APP_RESET) {
    state = undefined
  }

  return appReducer(state, action)
}
