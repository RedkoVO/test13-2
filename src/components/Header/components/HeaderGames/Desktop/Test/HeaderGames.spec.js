import * as React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'

import HeaderGames from '../'

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders correctly DESKTOP', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Wrapper>
            <HeaderGames children={<div>test</div>} />
          </Wrapper>
        </MemoryRouter>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
