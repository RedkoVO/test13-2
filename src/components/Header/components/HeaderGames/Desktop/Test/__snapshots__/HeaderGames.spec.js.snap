// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly DESKTOP 1`] = `
<div
  className="HeaderGames-root-1"
>
  <div
    className="HeaderGames-gamesUserBlock-2"
  >
    <p
      className="HeaderGames-gamesUserHello-3"
    >
       Hello, Ben A 
      <br />
      <a
        className="HeaderGames-gamesUserVerification-4"
        href="/"
        onClick={[Function]}
      >
        KYC VERIFICATION
      </a>
    </p>
    <div
      className="HeaderGames-gamesUserPhoto-5"
    >
      <img
        alt=""
        src="user.jpg"
      />
    </div>
  </div>
  <div>
    test
  </div>
</div>
`;
